core/main/src/broker/websocket_broker.rs

```diff
@@ -156,8 +161,22 @@ impl WSNotificationBroker {
         tokio::spawn(async move {
             let app_id = request_c.get_id();
             let alias = request_c.rule.alias.clone();
-            let (mut ws_tx, mut ws_rx) =
-                BrokerUtils::get_ws_broker(&url, Some(alias.clone())).await;
+            #[cfg(not(test))]
+            let config = WebSocketConfigBuilder::default().alias(alias).build();
+            #[cfg(test)]
+            let config = WebSocketConfigBuilder::default()
+                .alias(alias)
+                .retry(0)
+                .fail_after(1)
+                .build();
+
+            let resp = WebSocketUtils::get_ws_stream(&url, Some(config)).await;
+            if resp.is_err() {
+                error!("Error connecting to websocket broker");
+                tr.close();
+                return;
+            }
+            let (mut ws_tx, mut ws_rx) = resp.unwrap();
```

Comment:
unsafe `unwrap()`

Reply:
Not unsafe `if resp.is_err() {` is already checked